hal/src/electron/modem/dns_client.cpp

```diff
@@ -137,7 +138,9 @@ class UdpSocket {
     }
 
     int open() {
-        close();
+        if (sock_ >= 0) {
```

Comment:
What happens if the modem is power-cycled? The handle will no longer be valid or what's even worse we might be writing into the wrong socket handle.

Reply:
I'll make a change here based on our conversation this morning, adding a function the DNS Client can call in `mdm_hal.cpp` to know if the modem was power cycled.  If the modem is power cycled the DNS Client's `sock_` handle will be invalidated so that a new socket is opened.

Reply:
Ok, I pushed a commit with a bunch of improvements: automatic socket/handle cleanup on creation of sockets, detects state mismatch between system and modem, and reduces socket log noise [ch37610].